homeassistant/components/meteoclimatic/sensor.py

```diff
@@ -0,0 +1,109 @@
+    DOMAIN,
+    MANUFACTURER,
+    MODEL,
+    SENSOR_TYPE_CLASS,
+    SENSOR_TYPE_ICON,
+    SENSOR_TYPE_NAME,
+    SENSOR_TYPE_UNIT,
+    SENSOR_TYPES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_LAST_UPDATE = "last_update"
+ATTR_SENSOR_ID = "sensor_id"
+ATTR_STATION_CODE = "station_code"
+ATTR_STATION_NAME = "station_name"
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the Meteoclimatic sensor platform."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [MeteoclimaticSensor(sensor_type, coordinator) for sensor_type in SENSOR_TYPES],
+        False,
+    )
+
+
+class MeteoclimaticSensor(CoordinatorEntity, SensorEntity):
+    """Representation of a Meteoclimatic sensor."""
+
+    def __init__(self, sensor_type: str, coordinator: DataUpdateCoordinator) -> None:
+        """Initialize the Meteoclimatic sensor."""
+        super().__init__(coordinator)
+        self._type = sensor_type
+        station = self.coordinator.data["station"]
+        self._name = f"{station.name} {SENSOR_TYPES[self._type][SENSOR_TYPE_NAME]}"
+        self._unique_id = f"{station.code}_{self._type}"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the sensor."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self.platform.config_entry.unique_id)},
+            "name": self.coordinator.name,
+            "manufacturer": MANUFACTURER,
+            "model": MODEL,
+            "entry_type": "service",
+        }
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return (
+            getattr(self.coordinator.data["weather"], self._type)
+            if self.coordinator.data
+            else None
+        )
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return SENSOR_TYPES[self._type][SENSOR_TYPE_UNIT]
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return SENSOR_TYPES[self._type][SENSOR_TYPE_ICON]
+
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return SENSOR_TYPES[self._type][SENSOR_TYPE_CLASS]
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        attributes = {ATTR_ATTRIBUTION: ATTRIBUTION, ATTR_SENSOR_ID: self._type}
+        if self.coordinator.data:
+            attributes[ATTR_LAST_UPDATE] = self.coordinator.data["reception_time"]
+            attributes[ATTR_STATION_CODE] = self.coordinator.data["station"].code
+            attributes[ATTR_STATION_NAME] = self.coordinator.data["station"].name
+        return attributes
```

Comment:
Do we really need to add these attribute to every single sensor?

Reply:
It's not strictly necessary, but I thought it would be useful to have a reference to the station and the timestamp of the measurement. I can move that info to somewhere else, or get rid of it altogether if this is not considered appropriate.

Reply:
We generally try to reduce attributes as much as possible.
My main issue would be, that they apply to all the sensors added. So every sense will have the same station data in the attributes, which will be stored and logged each and every time anything changes to this entity.

Reply:
Got it - I've removed the extra attributes for now. I would like to keep a reference to the station code, but I'll think about how to do that without duplicating data.